### ISSUE TITLE
BOINC Monitoring daemon

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1474]
We need to create a new daemon that can be run with an elevated account.  The purpose of this daemon is to collect various pieces of information not available to the BOINC client when run from an unprivileged account.

Basically this daemon would start and periodically check for updated information like CPU Temperatures and whether or not virtualization extensions have been enabled in the BIOS or not.  All the information should be exposed to the BOINC client via memory mapped files.

Basically we will need access to the rdmsr CPU instruction which is generally restricted to executing in ring 0.

Windows References:
http://www.securiteam.com/windowsntfocus/5TP0B2KC0K.html (Use of ZwSystemDebugControl with DebugSysReadMsr)

Generic References:

Determine if the virtualization instruction set is enabled in the BIOS:
http://bazaar.launchpad.net/~cpu-checker-dev/cpu-checker/trunk/view/head:/kvm-ok

**Comment by CharlieFenton** [comment]
I added code to the client quite a long time ago which measures the temperature on Macs, but it has never actually been used, so it has not been tested recently. It is get_max_cpu_temperature() in hostinfo_unix.cpp lines 1004 - 1223 and does not require elevated privileges on the Mac.

Cheers,
--Charlie
